roborazzi-accessibility-check/src/main/java/com/github/takahirom/roborazzi/ATFAccessibilityChecker.kt

```diff
@@ -0,0 +1,137 @@
+      roborazziErrorLog("Skipping accessibilityChecks on API " + Build.VERSION.SDK_INT + "(< ${Build.VERSION_CODES.UPSIDE_DOWN_CAKE})")
+      return
+    }
+    // TODO remove this once ATF doesn't bail out
+    // https://github.com/google/Accessibility-Test-Framework-for-Android/blob/c65cab02b2a845c29c3da100d6adefd345a144e3/src/main/java/com/google/android/apps/common/testing/accessibility/framework/uielement/AccessibilityHierarchyAndroid.java#L667
+    ShadowBuild.setFingerprint("roborazzi")
+
+    if (captureRoot is CaptureRoot.Compose) {
+      val view =
+        (captureRoot.semanticsNodeInteraction.fetchSemanticsNode().root as ViewRootForTest).view.rootView
+
+      // Will throw based on configuration
+
+      val screenshot: Bitmap? =
+        RoboComponent.Compose(
+          node = captureRoot.semanticsNodeInteraction.fetchSemanticsNode(),
+          roborazziOptions = roborazziOptions
+        ).image
+
+      val results = runAllChecks(view, screenshot, checks)
+
+      reportResults(results, failureLevel)
+
+    } else if (captureRoot is CaptureRoot.View) {
+      val viewInteraction = captureRoot.viewInteraction
+      // Use perform to get the view
+      viewInteraction.perform(object : ViewAction {
+        override fun getDescription(): String {
+          return "Run accessibility checks"
+        }
+
+        override fun getConstraints(): Matcher<View> {
+          return Matchers.any(View::class.java)
+        }
+
+        override fun perform(uiController: UiController?, view: View?) {
+          if (view == null) {
+            throw IllegalStateException("View is null")
+          }
+          val results = runAllChecks(
+            view = view,
+            screenshotBitmap = RoboComponent.View(view, roborazziOptions).image,
+            checks = checks
+          )
+          reportResults(results, failureLevel)
+        }
+      })
+    }
+  }
+
+  private fun reportResults(
+    results: List<AccessibilityViewCheckResult>,
+    failureLevel: CheckLevel
+  ) {
+    // Report on any warnings in the log output if not failing
+    results.forEach { check ->
+      when (check.type) {
+        AccessibilityCheckResultType.ERROR -> roborazziErrorLog("Error: $check")
+        AccessibilityCheckResultType.WARNING -> roborazziErrorLog(
+          "Warning: $check"
+        )
+
+        AccessibilityCheckResultType.INFO -> roborazziReportLog(
+          "Info: $check"
+        )
+
+        else -> {}
+      }
+    }
+
+    val failures = results.filter { failureLevel.isFailure(it.type) }
+    if (failures.isNotEmpty()) {
+      throw AccessibilityViewCheckException(failures.toMutableList())
+    }
+  }
+
+  companion object
+}
+
+enum class CheckLevel(private vararg val failedTypes: AccessibilityCheckResultType) {
+  Error(AccessibilityCheckResultType.ERROR),
+
+  Warning(
+    AccessibilityCheckResultType.ERROR,
+    AccessibilityCheckResultType.WARNING
+  ),
+
+  LogOnly;
```

Comment:
We might need to implement `LogOnly`. I believe there is no implementation for this option.

Reply:
Sorry. I didn't read the code properly. It looks good.